### PR TITLE
Unlink file blocks in `file.del()`

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -161,8 +161,22 @@ module.exports = class RandomAccessFile extends RandomAccessStorage {
 
     function onstat (err, st) {
       if (err) return req.callback(err)
-      if (req.offset + req.size < st.size) return req.callback(null)
-      fs.ftruncate(fd, req.offset, ontruncate)
+
+      if (req.offset + req.size >= st.size) {
+        fs.ftruncate(fd, req.offset, ontruncate)
```
Comment (mafintosh) [inline index.js:166]: `return` after here so we can remove the else below

File: index.js [hunk 2]
```diff
@@ -161,8 +161,22 @@ module.exports = class RandomAccessFile extends RandomAccessStorage {
...
+      } else if (fsext) {
+        const i = Math.ceil(req.offset / st.blksize)
+        const j = Math.floor((req.offset + req.size) / st.blksize)
+
+        if (i >= j) return req.callback(null)
+
+        const offset = i * st.blksize
+        const size = (j - i) * st.blksize
+
+        fsext.trim(fd, offset, size).then(ontruncate, ontruncate)
```
Comment (mafintosh) [inline index.js:176]: return after here as well

**Comment by mafintosh** [comment]
@kasperisager 🔥🔥🔥🔥🔥🔥🔥

Should we add a test?

**Comment by kasperisager** [comment]
@mafintosh We definitely should! I'll also put back in draft state until https://github.com/hypercore-skunkworks/fs-native-extensions/pull/1 is released.

**Comment by kasperisager** [comment]
@mafintosh All set 👌

File: test/basic.js [hunk 3]
```diff
@@ -247,28 +247,49 @@ test('rmdir option with non empty parent', function (t) {
...
+  file.stat(function (err, st) {
+    t.absent(err, 'no error')
+    file.write(0, Buffer.alloc(st.blksize * 100), function (err) {
       t.absent(err, 'no error')
-      t.is(st.size, 100)
-      file.del(0, 40, function (err) {
+      file.stat(function (err, before) {
         t.absent(err, 'no error')
-        file.stat(function (err, st) {
+        file.del(st.blksize * 20, st.blksize * 50, function (err) {
```
[the remark this answers is not in the document]
Comment (kasperisager) [reply test/basic.js:286]: The existing deletion test targets partial blocks.

[the remark this answers is not in the document]
Comment (kasperisager) [reply test/basic.js:286]: We lack a test that does both though, will add.

[the remark this answers is not in the document]
Comment (mafintosh) [reply test/basic.js:286]: 👍

**Comment by mafintosh** [comment]
3.1.0

**Pull request merged**